### PR TITLE
Add another batch of updates to the changelog

### PR DESCRIPTION
**Pull request by ytmimi**

### CODE REVIEW

**Comment by calebcartwright** [comment]
thank you for this, happy to merge once conflicts are resolved

**Comment by ytmimi** [comment]
@calebcartwright want me to add entries for these PRs too?
* #5971
* #5582
* #6034
* #6073
* #6101
* #6085
* #6092
* #6112

**Comment by calebcartwright** [comment]
> @calebcartwright want me to add entries for these PRs too?
> 
>     * [Format diff line to be easily clickable #5971](https://github.com/rust-lang/rustfmt/pull/5971)
> 
>     * [Fix for #3805 - wrap macro that starts with nested body blocks #5582](https://github.com/rust-lang/rustfmt/pull/5582)
> 
>     * [Consolidating dependencies #6034](https://github.com/rust-lang/rustfmt/pull/6034)
> 
>     * [Fix #6069 #6073](https://github.com/rust-lang/rustfmt/pull/6073)
> 
>     * [fix clap deprecations #6101](https://github.com/rust-lang/rustfmt/pull/6101)
> 
>     * [Ensure space around binary exprs #6085](https://github.com/rust-lang/rustfmt/pull/6085)
> 
>     * [Fix failure with `=>` in comment after match `=>` #6092](https://github.com/rust-lang/rustfmt/pull/6092)
> 
>     * [Fix: ICE with expanded code #6105 #6112](https://github.com/rust-lang/rustfmt/pull/6112)

sure, i'll defer to you whether to add them here or in a separate PR. whichever would be easier for you

File: CHANGELOG.md
Comment (calebcartwright) [inline CHANGELOG.md:8]: Nit but should this be under the Changed section?

File: CHANGELOG.md
Comment (calebcartwright) [inline CHANGELOG.md:17]: Do you recall the associated issue or PR with this one?

File: CHANGELOG.md
Comment (calebcartwright) [inline CHANGELOG.md:53]: Do you know if there were any notable changes in this major version bump that would be worth mentioning here, or perhaps simply linking to (e.g. the `dirs` v5 release notes?)

File: CHANGELOG.md
Comment (ytmimi) [reply CHANGELOG.md:8]: Agreed. I think it would be more appropriate to list the `trace!` changes in the `Changed` section.

File: CHANGELOG.md
Comment (ytmimi) [reply CHANGELOG.md:53]: When I looked over the changes I didn't find anything that would impact rustfmt, but I think linking to the release notes would be a good idea.

**Comment by ytmimi** [comment]
@calebcartwright I made some changes. Take a look when you get a chance.

File: CHANGELOG.md
Comment (calebcartwright) [inline CHANGELOG.md:89]: Any concerns with this providing a significant change in behavior? Obviously we're deferring to a 3rd party crate for resolving global config files and I'd argue doing so for good reason, with a side effect being that behavior could change as we upgrade.

Even if some people desire the change behavior `5.0.1` provides, that doesn't mean it's not an impactful change

Comment (calebcartwright) [reply CHANGELOG.md:89]: Nevermind, we've history here: https://github.com/rust-lang/rustfmt/pull/5472

**Review by calebcartwright — APPROVED** [review]

**Pull request merged**
